src/preferences/PreferencesBase.js

```diff
@@ -173,6 +174,14 @@ define(function (require, exports, module) {
                 prefFile.read({}, function (err, text) {
                     if (err) {
                         if (createIfNew) {
+                            // Unreadable file is also unwritable -- delete so get recreated
+                            if (err === FileSystemError.NOT_READABLE || err === FileSystemError.UNSUPPORTED_ENCODING) {
+                                appshell.fs.moveToTrash(path, function (err) {
+                                    if (err) {
+                                        console.log("Cannot move unreadable preferences file to trash: " + path);
```

Comment:
I just realized that we talked about making this a modal dialog -- let me know if that's true or writing to console is ok.